### DISCUSSION TITLE
Support http proxy in GitRepository/HelmRepository

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#234] [Proposals]
If some of your sources need to go through a proxy while others do not, it could become difficult to manage the HTTP_PROXY / NO_PROXY env vars in the source-controller container appropriately. It would be useful to be able to specify a httpProxy directly in GitRepository/HelmRepository to avoid this.

**Comment by seaneagan** [comment]
... and ImageRepository

**Comment by hiddeco** [comment]
For the `GitRepository` type, you will likely run into the same shared runtime HTTP client issues as https://github.com/fluxcd/source-controller/issues/93#issuecomment-661219491.

For the `HelmRepository` (and `HelmChart`) type, this will likely require [upstream changes to the `httpgetter`](https://github.com/helm/helm/blob/b18e7e201e55ba38fdecbcf81e4da512e55444b4/pkg/getter/httpgetter.go#L92).

**Reply by stefanprodan** [reply]
Yeah I see no way around the shared HTTP client in go-git, we need to fork it and try to port the changes upstream.

**Comment by gberche-orange** [comment]
The workaround of adding the http_proxy,https_proxy,no_proxy environment variables to the source controller deployment is not solving the use-case of selecting different http proxies per GitRepository/HelmRepository.

In our company, the workloads running on public cloud platforms need to select a distinct http_proxy for intranet access and internet access. This currently prevents some usage of GitRepository that require the non default http proxy.
